File: backend/models/prompts/folders.py
# models/prompts/folders.py
from pydantic import BaseModel
from typing import Optional, Dict
from enum import Enum

class FolderBase(BaseModel):
    id: Optional[int] = None
    title: str
    
class FolderCreate(FolderBase):
    description: str = None
    user_id: Optional[str] = None
    organization_id: Optional[str] = None
    company_id: Optional[str] = None
    parent_folder_id: Optional[int] = None
    type: Optional[str] = None
    
class FolderUpdate(FolderCreate):
    pass

class FolderResponse(FolderBase):
    id: int
    title: str
    description: Optional[str] = None
    user_id: Optional[str] = None
    organization_id: Optional[str] = None
    company_id: Optional[str] = None
    parent_folder_id: Optional[int] = None

File: frontend/types/prompts/folders.ts
// src/types/prompts/folders.ts
import { Template } from './templates';

/**
   * Template folder structure
   */
export interface TemplateFolder {
  id: number;
  title?: string;
  description?: string;
  type: 'company' | 'organization' | 'user';
  company_id?: number;
  organization_id?: number;
  user_id?: string;
  templates: Template[];
  Folders?: TemplateFolder[];
  is_pinned?: boolean;
  parent_folder_id?: number | null;
  created_at?: string;
  updated_at?: string;
}
